src/service.rs

```diff
@@ -0,0 +1,57 @@
+use Body;
+
+use http::{Request, Response};
+use futures::{Future, Poll};
+use tower::Service;
+
+/// An HTTP service
+///
+/// This is not intended to be implemented directly. Instead, it is a trait
+/// alias of sorts. Implements the `tower::Service` trait using `http::Request`
+/// and `http::Response` types.
+pub trait HttpService: ::sealed::Sealed {
```

Comment:
Should this maybe be `Http2Service`, since it requires the bodies to implement `tower_h2::Body`?

Reply:
Well, it is in the tower-h2 crate, so my thought was that HTTP/2.0 was assumed.